### PR TITLE
Join Room Refactor

### PR DESCRIPTION
**Pull request by o4codes**
Just a few code refactorings was made:
-- removed add_to_room mode because it wasn't reusable and would typically be used within just the join room function.
-- used fastapi body function parameter instead
-- used a dict comprehension to convert objects of room member to dictionary in order to serialize the objects of the class to JSON.

### CODE REVIEW

**Review by zxenonx — APPROVED** [review]
Thanks for the work.
LGTM.

File: backend/endpoints/rooms.py [hunk 1]
```diff
@@ -114,15 +117,18 @@ async def add_to_room(
...
         HTTP_424_FAILED_DEPENDENCY: failed to add new members to room
     """
-
     DB = DataStorage(org_id)  # initializes the datastorage class with the org id
-    new_member = data.dict().get("new_member")
+
+    members = {
+        k: v.dict() for k, v in new_members.items()
+    }  # converts RoomMember to dict
+
```
Comment (mikenrowland) [inline backend/endpoints/rooms.py:125]: Respect boss. 🙌🏽

Comment (mikenrowland) [reply backend/endpoints/rooms.py:125]: Great work.

**Review by mikenrowland — APPROVED** [review]

File: backend/endpoints/rooms.py (hunk 1 above)
Comment (o4codes) [reply backend/endpoints/rooms.py:125]: You did the main work man.
Thanks.

**Comment by o4codes** [comment]
> Thanks for the work. LGTM.

Thanks man

**Pull request merged**